src/main/java/frc/trigon/robot/utilities/SpeedMechanism2d.java

```diff
@@ -0,0 +1,118 @@
+        createLigaments(velocity, targetVelocity);
+        setTargetVelocity(targetVelocity);
+        setTargetVelocityArrowAngle(targetVelocity);
+        targetVelocityLigament.setLength(targetVelocity);
+        setCurrentVelocityArrowAngle(velocity);
+        currentVelocityLigament.setLength(velocity);
+        setLigamentColor(velocity);
+        Logger.recordOutput(key, mechanism);
+    }
+
+    public void setTargetVelocity(double targetVelocity) {
+        lastTargetVelocity = targetVelocity;
+    }
+
+    private void createLigaments(double currentVelocity, double targetVelocity) {
+        String currentVelocityLigamentKeyZ = "";
+        String targetVelocityLigamentKeyZ = "";
+        if (isCurrentVelocityLigamentBiggerThanTargetVelocityLigament(currentVelocity, targetVelocity))
+            targetVelocityLigamentKeyZ = "Z";
+        else
+            currentVelocityLigamentKeyZ = "Z";
+        this.currentVelocityLigament = root.append(new MechanismLigament2d(currentVelocityLigamentKeyZ + "CurrentVelocityLigament", 0, 0, 5, new Color8Bit(Color.kBlue)));
+        this.currentVelocityTopArrowLigament = currentVelocityLigament.append(new MechanismLigament2d(currentVelocityLigamentKeyZ + "CurrentVelocityTopArrowLigament", 0.2 * maxDisplayableVelocity, zeroTopAngle, 5, new Color8Bit(Color.kBlue)));
+        this.currentVelocityBottomArrowLigament = currentVelocityLigament.append(new MechanismLigament2d(currentVelocityLigamentKeyZ + "CurrentVelocityBottomArrowLigament", 0.2 * maxDisplayableVelocity, zeroBottomAngle, 5, new Color8Bit(Color.kBlue)));
+
+        this.targetVelocityLigament = root.append(new MechanismLigament2d(targetVelocityLigamentKeyZ + "TargetVelocityLigament", 0, 0, 5, new Color8Bit(Color.kGray)));
+        this.targetVelocityTopArrowLigament = targetVelocityLigament.append(new MechanismLigament2d(targetVelocityLigamentKeyZ + "TargetVelocityTopArrowLigament", 0.2 * maxDisplayableVelocity, zeroTopAngle, 5, new Color8Bit(Color.kGray)));
+        this.targetVelocityBottomArrowLigament = targetVelocityLigament.append(new MechanismLigament2d(targetVelocityLigamentKeyZ + "TargetVelocityBottomArrowLigament", 0.2 * maxDisplayableVelocity, zeroBottomAngle, 5, new Color8Bit(Color.kGray)));
+    }
+
+    private boolean isCurrentVelocityLigamentBiggerThanTargetVelocityLigament(double currentVelocity, double targetVelocity) {
+        if (currentVelocity > deadband && targetVelocity > deadband)
+            return currentVelocity > targetVelocity;
+        else if (currentVelocity < -deadband && targetVelocity < -deadband)
+            return currentVelocity < targetVelocity;
+        return false;
+    }
+
+    private void setLigamentColor(double velocity) {
+        currentVelocityLigament.setColor(velocityToColor(velocity));
+        currentVelocityTopArrowLigament.setColor(velocityToColor(velocity));
+        currentVelocityBottomArrowLigament.setColor(velocityToColor(velocity));
+    }
```

Comment:
1. Rename this to setCurrentLigamnetColor
2. Make this accept the color, not the vel.